### ISSUE TITLE
Add ExceptT to Ricochet Monad transformer stack

### ISSUE THREAD

**Issue by photm5** [Jugendhackt/haskell-ricochet#16]
We’ll want to respond to errors, eg. see [Connection.hs](https://github.com/Jugendhackt/haskell-ricochet/blob/master/src/Network/Ricochet/Connection.hs).

**Comment by froozen** [comment]
Where exactly do we need exceptions?

**Comment by photm5** [comment]
Whenever the remote peer sends something we cannot parse. I opened this issue when there were stdout messages in Connection.hs.

**Comment by sternenseemann** [comment]
Would `fail` be enough for now?

**Comment by froozen** [comment]
No, because usually `fail = error`. That would cause a crash, which is not really what we want.

**Comment by sternenseemann** [comment]
I see.
